File: pope_dogc/pope_dog.c
#include <stdio.h>
#ifdef _WIN32
#include <Windows.h>
#else
#include <unistd.h>
#endif
const int pollingDelay = 100;

void print_poem(){
    
    printf("у попа была собака, он её любил\n");
    printf("она съела кусок мяса, он её убил\n");
    printf("в землю закопал\n");
    printf("и надпись написал, что ");
    
    //sleep:
    #ifdef _WIN32
    Sleep(pollingDelay);
    #else
    usleep(pollingDelay*1000);  /* sleep for 100 milliSeconds */
    #endif
    
    print_poem();
   
}


int  main(){
    print_poem();
    return 0;
}

File: pope_dogjs/pope_dog.js
function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
  }

 async function print_poem(){
    console.log("у попа была собака, он её любил");
    console.log("она съела кусок мяса, он её убил");
    console.log("в землю закопал");
    console.log("и надпись написал, что ");
    await sleep(2000);
    print_poem();

}

print_poem();